### PR TITLE
Add init container documentation

### PR DESCRIPTION
**Pull request by smarterclayton**
Resolves kubernetes/kubernetes#1589

### CODE REVIEW

**Comment by erictune** [comment]
@caesarxuchao 
I think this feature is new to 1.3.  Please close this PR, and rebase your work on top of release-1.3, and open a new PR.

To rebase your work on 1.3 do something like this starting in the branch that you used for this PR:
```
COMMITS=$(git log master.. | grep commit | cut -f 2 -d " " | tac)
BR=$(git rev-parse --abbrev-ref HEAD)
git fetch upstream
git checkout upstream/release-1.3
git checkout -b $BR-1.3
for c in $COMMITS
do
  git cherry-pick $c; done
done
git push
```

**Comment by erictune** [comment]
Never mind, you already did this.

**Comment by pwittrock** [comment]
@smarterclayton do we expect the usage to change a lot come beta?

**Comment by smarterclayton** [comment]
Expect?  No.  Maybe some tweaks to which features are supported, and the fact that it would not be using annotations.

**Comment by pwittrock** [comment]
@smarterclayton The annotations was mainly what I was asking about.

**Comment by johndmulhausen** [comment]
Excellent work, thanks @smarterclayton ! And thanks for the tech review, @pwittrock :)

File: docs/user-guide/production-pods.md [hunk 1]
```diff
@@ -180,6 +180,26 @@ Other times, applications are only temporarily unable to serve, and will recover
...
+* Performing configuration templating from environment variables into a config file
+* Registering the pod into a central database, or fetching remote configuration from that database
+* Downloading application dependencies, seed data, or preconfiguring disk
+
+Kubernetes now includes an alpha feature known as **init containers**, which are one or more containers in a pod that get a chance to run and initialize shared volumes prior to the other application containers starting.  An init container is exactly like a regular container, except that it always runs to completion and each init container must complete successfully before the next one is started. If the init container fails (exits with a non-zero exit code) on a `RestartNever` pod the pod will fail - otherwise it will be restarted until it succeeds or the user deletes the pod.
+
+Since init containers are an alpha feature, they are specified by setting the `pod.alpha.kubernetes.io/init-containers` annotation on a pod (or replica set, deployment, daemon set, pet set, or job). The value of the annotation must be a string containing a JSON array of container definitions:
```
Comment (gtaylor) [inline docs/user-guide/production-pods.md]: > string containing a JSON array of container definitions

I don't suppose it also supports YAML?

**Pull request merged**